packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m

```diff
@@ -0,0 +1,244 @@
+    NSArray *singleOrientations = @[@(UIInterfaceOrientationMaskPortrait),
+                                    @(UIInterfaceOrientationMaskPortraitUpsideDown),
+                                    @(UIInterfaceOrientationMaskLandscapeLeft),
+                                    @(UIInterfaceOrientationMaskLandscapeRight)];
+    // If the particular orientation is supported, we add it to the array of allowedOrientations
+    NSMutableArray *allowedOrientations = [[NSMutableArray alloc] init];
+    for (NSNumber *wrappedSingleOrientation in singleOrientations) {
+      UIInterfaceOrientationMask singleOrientationMask = [wrappedSingleOrientation intValue];
+      UIInterfaceOrientationMask supportedOrientation = orientationMask & singleOrientationMask;
+      if (supportedOrientation == singleOrientationMask){
+        EXOrientation orientation = [EXScreenOrientationUtilities orientationNativeToJS: (UIInterfaceOrientationMask) singleOrientationMask];
+        [allowedOrientations addObject:[EXScreenOrientationUtilities orientationToString:orientation]];
+      }
+    }
+    resolve([allowedOrientations copy]);
+}
+
+UM_EXPORT_METHOD_AS(doesSupportAsync,
+                    doesSupportAsync:(NSString *)orientationLockString
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  [self supportsOrientationLockAsync:orientationLockString resolver:resolve rejecter:reject];
+}
+
+UM_EXPORT_METHOD_AS(supportsOrientationLockAsync,
+                    supportsOrientationLockAsync:(NSString *)orientationLockString
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  EXOrientationLock orientationLock = [EXScreenOrientationUtilities stringToOrientationLock:orientationLockString];
+  UIInterfaceOrientationMask orientationMask = [EXScreenOrientationUtilities orientationLockJSToNative:orientationLock];
+  if (orientationMask == INVALID_MASK) {
+    resolve(@NO);
+  } else if ([EXScreenOrientationUtilities doesSupportOrientationMask:orientationMask]) {
+    resolve(@YES);
+  } else {
+    resolve(@NO);
+  }
+}
+
+UM_EXPORT_METHOD_AS(getOrientationAsync,
+                    getOrientationAsyncResolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  resolve([EXScreenOrientationUtilities UIInterfaceOrientationToEXOrientation:_currentScreenOrientation]);
+}
+
+- (UIInterfaceOrientationMask)orientationMask
+{
+  return [[EXScreenOrientationModule sharedRegistry] orientationMaskForAppId:defaultAppId];
+}
+
+- (void)setOrientationMask:(UIInterfaceOrientationMask)mask
+{
+  return [[EXScreenOrientationModule sharedRegistry] setOrientationMask:mask forAppId:defaultAppId];
+}
+
++ (EXScreenOrientationRegistry *)sharedRegistry
+{
+  return (EXScreenOrientationRegistry *)[UMModuleRegistryProvider getSingletonModuleForClass:[EXScreenOrientationRegistry class]];
```

Comment:
Do not use `forClass` anywhere inside unimodules — use `forName` otherwise versioned code will end up using versioned registry